app/controllers/publishers_controller.rb

```diff
@@ -139,15 +173,34 @@ def authenticate_via_token
     sign_out(current_publisher) if current_publisher
     return if params[:id].blank? || params[:token].blank?
     publisher = Publisher.find(params[:id])
-    if publisher.authentication_token.present? && publisher.authentication_token == params[:token]
+    if PublisherTokenAuthenticator.new(publisher: publisher, token: params[:token]).perform
       sign_in(:publisher, publisher)
+    else
+      flash[:alert] = I18n.t("publishers.authentication_token_invalid")
     end
   end
 
+  # Used by #create_auth_token
+  # Kinda copied from Publisher #normalize_brave_publisher_id
+  def normalize_brave_publisher_id(brave_publisher_id)
+    require "faraday"
+    PublisherDomainNormalizer.new(domain: brave_publisher_id).perform
+  rescue PublisherDomainNormalizer::DomainExclusionError
+    "#{I18n.t('activerecord.errors.models.publisher.attributes.brave_publisher_id.exclusion_list_error')} #{Rails.application.secrets[:support_email]}"
+  rescue PublisherDomainNormalizer::OfflineNormalizationError => e
+    e.message
```

Comment:
if these errors are uncaught it will result in a 500, for which there's a [catch-all error page](https://publishers.brave.com/blabla404).